modules/extensions-support/src/main/java/org/mule/module/extension/internal/introspection/AnnotationsBasedDescriber.java

```diff
@@ -306,8 +304,35 @@ private void addTypeRestrictions(ParameterDescriptor parameter, org.mule.module.
         }
     }
 
-    private void describeCapabilities(DeclarationDescriptor declaration, Class<?> extensionType)
+    private void declareExtensionCapabilities(DescribingContext context, DeclarationDescriptor declaration, Class<?> extensionType)
+    {
+        declareCapabilities(context, declaration, extractor -> extractor.extractExtensionCapability(declaration, extensionType));
+    }
+
+    private void declareConfigCapabilities(DescribingContext context, ConfigurationDescriptor descriptor, Class<?> configType)
     {
-        capabilitiesResolver.resolveCapabilities(declaration, extensionType);
+        declareCapabilities(context, descriptor, extractor -> extractor.extractConfigCapability(descriptor, configType));
+    }
+
+    private void declareOperationCapabilities(DescribingContext context, OperationDescriptor descriptor, Method method)
+    {
+        declareCapabilities(context, descriptor, extractor -> extractor.extractOperationCapability(descriptor, method));
+    }
+
+    private void declareCapabilities(DescribingContext context, HasCapabilities<?> declaration, Function<CapabilityExtractor, Object> extractFunction)
+    {
+        List<CapabilityExtractor> extractors = context.getParameter(CAPABILITY_EXTRACTORS, List.class);
+        if (CollectionUtils.isEmpty(extractors))
+        {
+            return;
+        }
+
+        extractors.forEach(extractor -> {
```

Comment:
my thoughts exactly, but I think that should be part of the general reformatting of the code. I don't think there's much value in tackling this problem on its own.

Meanwhile, what I can tell you is that these lambdas are intellij formatted so at the very least we know they will be consistent across team members